src/services/helpers.ts

```diff
@@ -9,14 +10,51 @@ export class ValidationError extends Error {
   }
 }
 
+export class TimeoutError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "TimeoutError";
+  }
+}
+
+async function processTimeout<T>(
+  ms: TimeoutMS,
+  promise: Promise<T>,
+  errorMessage?: string
+): Promise<T> {
+  return new Promise<T>(async (resolve, reject) => {
+    setTimeout(() => {
+      reject(new TimeoutError(errorMessage || "Request timed out"));
+    }, ms);
+    resolve(await promise);
+  });
+}
+
+// If timeout
+// true => timeout duration for fetch request will be set as env var.
+// false => usual fetch request is done.
+// TimeoutMS (eg. 10) => timeout duration for fetch request will be set as given val in milliseconds.
```

Comment:
Btw you can do something like this instead: 
![image](https://user-images.githubusercontent.com/577802/87644282-833f6c80-c77e-11ea-8ec6-634cb747122c.png)

Then it will appear nicely in the intellisense:
![image](https://user-images.githubusercontent.com/577802/87644341-96ead300-c77e-11ea-848e-433c9a1c87e6.png)

Reply:
@dhemutton don't forget this change too 😆